api/src/main/java/org/openmrs/DrugOrder.java

```diff
@@ -477,27 +482,27 @@ public void setAutoExpireDateBasedOnDuration() {
 			setAutoExpireDate(getDosingInstructionsInstance().getAutoExpireDate(this));
 		}
 	}
-	
+
 	public String toString() {
 		String prefix = DISCONTINUE == getAction() ? "DC " : "";
 		return prefix + "DrugOrder(" + getDose() + getDoseUnits() + " of "
-		        + (getDrug() != null ? getDrug().getName() : "[no drug]") + " from " + getDateActivated() + " to "
+		        + (getDrugNonCoded() != null ? getDrugNonCoded() : (getDrug() != null ? getDrug().getName() : "[no drug]")) + " from " + getDateActivated() + " to "
```

Comment:
Is it a good idea to use service in the DrugOrder model..? The reason we proceeded with just nonCoded value is because the concept validations are already in DrugOrderValidator.